components/21-atoms/table/table-mobile.functions.js

```diff
@@ -0,0 +1,167 @@
+      // Get the headers of the table columns.
+      let colHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="col"');
+      // Get the headers of the table rows.
+      let rowHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="row"]');
+      // Get all rows of the table.
+      let rowsNodeList = element.querySelector('table').querySelectorAll('tr');
+      // The first row of the table.
+      const firstRow = element.querySelector('tr');
+
+      // Determine if we have column headings.
+      options.columnHeaders = (firstRow.getElementsByTagName('th').length === firstRow.querySelectorAll('*').length);
+
+      // Determine if we have column headers. If so set the initial row +1.
+      let initial = (options.columnHeaders) ? 1 : 0;
+
+      // Add the rows as a list item.
+      for (let i = initial; i < rowsNodeList.length; i++) {
+        let row = rowsNodeList[i];
+
+        // Get all the columns of the table based on the current row.
+        let colsNodeList = row.querySelectorAll('td');
+
+        let listItem = document.createElement('li');
+        list.appendChild(listItem);
+
+        // Add a header based on the row heading.
+        if (rowHeadingsNodeList !== undefined) { // eslint-disable-line no-undefined
+          let header = document.createElement(options.headingType);
+          header.innerHTML = row.querySelector('th[scope="row"]').innerHTML;
+          listItem.appendChild(header);
+        }
+
+        // Create the definition list containing all columns as dd's.
+        let defList = document.createElement('dl');
+        listItem.appendChild(defList);
+
+        for (let j = 0; j < colsNodeList.length; j++) {
+          let column = colsNodeList[j];
+
+          if (colHeadingsNodeList.length > 0) {
+            let term = document.createElement('dt');
+            term.innerHTML = colHeadingsNodeList[j + 1].innerHTML; // Refactor this j + 1 line.
+            defList.appendChild(term);
+          }
+
+          let item = document.createElement('dd');
+          item.innerHTML = column.innerHTML;
+          defList.appendChild(item);
+        }
+      }
+    };
+
+    /**
+     * Add an unordered list element to the table-list container.
+     */
+    const addList = () => {
+      let list = document.createElement('ul');
+      tableList.appendChild(list);
+      addListItems(list);
+
+      addListDescription();
+    };
+
+    /**
+     * Add the table caption as a description to the definition lists.
+     */
+    const addListDescription = () => {
+      let description = document.createElement('div');
+      description.setAttribute('class', 'list-description');
+      description.setAttribute('id', 'list-description');
+      description.innerHTML = caption.innerHTML;
+      tableList.appendChild(description);
+    };
+
+    /**
+     * Build a mobile version of the table with definition lists.
+     */
+    const buildMobileTable = () => {
```

Comment:
This function should be removed. It does nothing but call another function with no parameters.

Reply:
OCD! :D